test/e2e/framework/exec.go

```diff
@@ -68,10 +63,10 @@ func (f *Framework) ExecWithOptions(options ExecOptions, index ClusterIndex) (st
 	}
 
 	if options.PreserveWhitespace {
-		return stdout.String(), stderr.String(), err
+		return stdout.String(), stderr.String(), nil
```

Comment:
why discard error? what if we fail even after 5 attempts, we need to return error.